pkg/handlers/generic/lifecycle/utils/utils.go

```diff
@@ -105,12 +105,18 @@ func EnsureNamespace(ctx context.Context, c ctrlclient.Client, name string) erro
 		},
 	}
 
-	// check if namespace exists and return early if it does
-	if err := c.Get(ctx, ctrlclient.ObjectKeyFromObject(ns), ns); err == nil {
-		return nil
-	}
+	return EnsureNamespace(ctx, c, ns)
+}
 
-	err := client.ServerSideApply(ctx, c, ns, client.ForceOwnership)
+// EnsureNamespace will create the namespace if it does not exist.
+func EnsureNamespace(ctx context.Context, c ctrlclient.Client, ns *corev1.Namespace) error {
+	if ns.TypeMeta.APIVersion == "" {
+		ns.TypeMeta.APIVersion = corev1.SchemeGroupVersion.String()
+	}
+	if ns.TypeMeta.Kind == "" {
+		ns.TypeMeta.Kind = "Namespace"
+	}
```

Comment:
nit: we can use Go 1.22 magic :) 
```suggestion
	ns.TypeMeta.APIVersion = cmp.Or(ns.TypeMeta.APIVersion, corev1.SchemeGroupVersion.String())
	ns.TypeMeta.Kind = cmp.Or(ns.TypeMeta.APIVersion, "Namespace")
```

Reply:
Oh I like this!

Reply:
I appreciate the suggestion. It is more concise, but I find it harder to understand, because you have to (a) know `cmp.Or` semantics, and (b) the zero values of the fields.

Apparently, `cmp.Or` [originated](https://blog.carlana.net/post/2024/golang-cmp-or-uses-and-history/) with a proposal for a `strings.First` function, which seems easier to understand.